ols/app/models/config.py

```diff
@@ -5,580 +5,213 @@
-        self.name = data.get("name", None)
-        self.url = data.get("url", None)
-        self.credentials = _get_attribute_from_file(data, "credentials_path")
-        if "models" not in data or len(data["models"]) == 0:
-            raise InvalidConfigurationError(
-                f"no models configured for provider {data['name']}"
-            )
-        for m in data["models"]:
-            if "name" not in m:
-                raise InvalidConfigurationError("model name is missing")
-            model = ModelConfig(m)
-            self.models[m["name"]] = model
-
-    def __eq__(self, other) -> bool:
-        """Compare two objects for equality."""
-        if isinstance(other, ProviderConfig):
-            return (
-                self.name == other.name
-                and self.url == other.url
-                and self.credentials == other.credentials
-                and self.models == other.models
-            )
-        return False
-
-    def validate_yaml(self) -> None:
-        """Validate provider config."""
-        if self.name is None:
-            raise InvalidConfigurationError("provider name is missing")
-        if self.url is not None and not _is_valid_http_url(self.url):
-            raise InvalidConfigurationError(
-                "provider URL is invalid, only http:// and https:// URLs are supported"
-            )
-        for v in self.models.values():
-            v.validate_yaml()
-
-
-class LLMProviders(BaseModel):
-    """LLM providers configuration."""
+    name: str # duplicates
+    # url??
+
+class LLMProviderConfig(BaseModel):
+    url: str
+    credentials_path: str # ???
+    models: list[ModelConfig]
+    user: str = None
+    password: str = None
+
+    @field_validator("url")
+    def validate_url(cls, v):
+        if not _is_valid_http_url(v):
+            raise ValueError(f"URL {v} is invalid")
+        return v
+
+    def __init__(self, **data):
+        super().__init__(**data)
+        self.user = _get_attribute_from_file(self.credentials_path) # ???
+        self.password = _get_attribute_from_file(self.credentials_path) # ???
 
-    providers: dict[str, ProviderConfig] = {}
-
-    def __init__(self, data: Optional[dict] = None) -> None:
-        """Initialize configuration and perform basic validation."""
-        super().__init__()
-        if data is None:
-            return
-        for p in data:
-            if "name" not in p:
-                raise InvalidConfigurationError("provider name is missing")
-            provider = ProviderConfig(p)
-            self.providers[p["name"]] = provider
-
-    def __eq__(self, other) -> bool:
-        """Compare two objects for equality."""
-        if isinstance(other, LLMProviders):
-            return self.providers == other.providers
-        return False
-
-    def validate_yaml(self) -> None:
-        """Validate LLM config."""
-        for v in self.providers.values():
-            v.validate_yaml()
+    def __eq__(self, other):
```

Comment:
Pydantic implements equality magic on its own. It is not sufficient enough?
What is the use case anyway for comparing two models/configs?